src/main/java/org/springframework/data/couchbase/core/RxJavaCouchbaseTemplate.java

```diff
@@ -66,7 +68,8 @@
     private final TranslationService translationService;
     private Consistency configuredConsistency = Consistency.DEFAULT_CONSISTENCY;
     private WriteResultChecking writeResultChecking = DEFAULT_WRITE_RESULT_CHECKING;
-
+    private Validator validator;
```

Comment:
Introducing `Validator` at Template API level requires the Validation API to be present on the class path of each project that wants to use Spring Data Couchbase. We cannot do that.

Instead, the validation should live in an optional component such as a validation event listener. Take a look at [Mongo's `ValidatingMongoEventListener`](https://github.com/spring-projects/spring-data-mongodb/blob/master/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/mapping/event/ValidatingMongoEventListener.java).